test/unit/error.spec.ts

```diff
@@ -292,3 +285,101 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+      try {
+          throw { x: "Hello error object!" };
+      } catch (error) {
+          return error.x;
+      }
+    `.expectToMatchJsResult();
+});
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError", "MyError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            class MyError extends Error {
```

Comment:
This looks strange, split this into two tests -> throw builtin errors as classes & throw custom Error as class (does this already exist as test 'subclass error'?)

Reply:
You're right, I was misusing the test to check if the string conversion works for subclasses. I've extended the `subclass error` test.